text/0000-postgres-major-upgrade.md

```diff
@@ -0,0 +1,58 @@
+- Feature Name: `PostgreSQL major version upgrade`
+- Start Date: 2023-09-30
+- RFC PR: (leave this empty)
+- Bundler Issue: (leave this empty)
+
+# Summary
+
+A structured four-phase approach to upgrade the PostgreSQL database for RubyGems.org from version 11 to version 12 with minimal downtime.
+
+# Motivation
+
+The primary goal is to ensure RubyGems.org benefits from the latest features, performance improvements, and security patches in PostgreSQL 12. Furthermore, by adopting this systematic approach, we lay down a foundation for streamlined future database upgrades, ensuring consistent minimal disruptions. Currently running PostgreSQL 11 reaches EOL on RDS at February 2024.
+
+# Guide-level explanation
+
+The PostgreSQL upgrade is organized into four distinct phases:
+
+1. **[Preparations](https://github.com/rubygems/rfcs/pull/53)**: 
+   - Set up the necessary tools and configurations, including initializing a PostgreSQL 12 RDS instance and deploying `pgbouncer` in Kubernetes.
+   - Configure the current PostgreSQL 11 RDS instance to support logical replication, accounting for the brief downtime required to enable this feature.
+   - Initiate logical replication from the PostgreSQL 11 instance to the newly created PostgreSQL 12 instance, ensuring data synchronization begins.
+  
+2. **Sanity Checks**: 
+   - Test and verify logical replication.
+   - Ensure `pgbouncer` functionalities are working as expected.
+   - Simulate and test the effects of `goodjob` worker shutdowns, pgbouncer pool pausing and measuring time needed to get in full sync.
+  
+3. **Final Migration**: 
+   - Pause `goodjob` workers.
+   - Confirm synchronization between the PostgreSQL instances including manual synchronization of sequences.
+   - Switch to the PostgreSQL 12 instance.
+   - Deploy using direct connection to resume `goodjob` workers.
+
+4. **Cleanup**: 
+   - Decommission outdated resources.
+   - Ensure backups target the new PostgreSQL 12 instance.
```

Comment:
in (1), we should backup both 11 and 12 RDS instances

Reply:
added